### ISSUE TITLE
Feature to search for symbols in files/workspace (like LSP?)

### ISSUE THREAD

**Issue by cxxxr** [lem-project/lem#924]

**Comment by seanfarley** [comment]
> Maybe we can use detective for this, expanding the capabilities to work with workspace files also maybe use this oportunity to make detective multi-thread and more efficient 👍

It seems to me to make more sense to write this symbol searching for lsp and then write a lsp-detective bridge so that we can have one api surface here.

**Comment by Sasanidas** [comment]
Yes, that is what I had in mind, I was talking when LSP is not available (when is available, use it instead). 

LSP workspaces symbols should be easy to implement, the server already give you everything, you just need to plug in the interface.
